x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_cleanup.tsx

```diff
@@ -7,7 +7,29 @@
 import createContainer from 'constate-latest';
 import { useMemo } from 'react';
 import { useTrackedPromise } from '../../../utils/use_tracked_promise';
-import { callDeleteJobs, callStopDatafeed } from './api/ml_cleanup';
+import { callDeleteJobs, callStopDatafeed, callGetJobDeletionTasks } from './api/ml_cleanup';
+import { getAllModuleJobIds } from '../../../../common/log_analysis';
+
+const timeout = (ms: number) => new Promise(res => setTimeout(res, ms));
+
+const waitUntilJobsAreDeleted = async (spaceId: string, sourceId: string) => {
+  while (true) {
+    const response = await callGetJobDeletionTasks();
+    const jobIdsBeingDeleted = response.jobIds;
+    const moduleJobIds = getAllModuleJobIds(spaceId, sourceId);
+    const needToWait = jobIdsBeingDeleted.some(jobId => moduleJobIds.includes(jobId));
+    if (needToWait) {
+      await timeout(1000);
+    } else {
+      return true;
+    }
+  }
+};
+
+const deleteJobs = async (spaceId: string, sourceId: string) => {
+  await callDeleteJobs(spaceId, sourceId);
+  return await waitUntilJobsAreDeleted(spaceId, sourceId);
```

Comment:
By returning the result of `await waitUntilJobsAreDeleted()` instead of await `callDeleteJobs()` as before the return type of `cleanupMLResources()` changed from a promise of a map of deleted jobs to a single boolean. Maybe we want to preserve that in case we want to perform some kind of consistency checking later on?

Reply:
Yep, good point. As nothing was consuming the response at this point I was a bit haphazard with it. I've changed it to return the response from the `callDeleteJobs()` call so it's more useful going forward.